src/Stryker.Core/Stryker.Core.UnitTest/InjectedHelpers/InjectedHelperTests.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Pipes;
+using System.Linq;
+using System.Text;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Shouldly;
+using Stryker.Core.Initialisation;
+using Stryker.Core.InjectedHelpers;
+using Stryker.Core.MutationTest;
+using Xunit;
+
+namespace Stryker.Core.UnitTest.InjectedHelpers
+{
+    public class InjectedHelperTests
+    { 
+        [Fact]
+        public void InjectHelpers_ShouldCompile()
+        {
+            var assemblies = AppDomain.CurrentDomain.GetAssemblies();
+            // it would be good to be ensure those assemblies are referenced by the build project
+            var needed = new[] {".CoreLib", ".Runtime", "System.IO.Pipes", ".Collections", ".Console" };
+            var references = new List<MetadataReference>();
+            var hack = new NamedPipeClientStream("test");
+            foreach (var assembly in assemblies)
+            {
+                if (needed.Any(x => assembly.FullName.Contains(x)))
+                {
+                    references.Add(MetadataReference.CreateFromFile(assembly.Location));
+                }
+            }
+
+            var compilation = CSharpCompilation.Create("dummy.dll",
+                CodeInjection.MutantHelpers,
+                options: new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary),
+                references: references);
+
+            // first try compiling
+            using (var ms = new MemoryStream())
+            {
+                // reset the memoryStream
+                var emitResult = compilation.Emit(
```

Comment:
This is a very heavy operation! Maybe we can find some way to achieve this without actually compiling the files? I've been struggling with this myself as well in the Compilation tests. Maybe checking for static errors is enough?

Reply:
I modified the test using the GetDiagnostics method instead of Emit, which divide the test time by around 2.5 (5 seconds vs 12 on my machine).
Not sure it can get any faster